lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -506,6 +506,7 @@ tokenize(String, Line, Column, Scope, Tokens) ->
   case tokenize_identifier(String, Line, Scope) of
     {Kind, Atom, Rest, Length, Ascii, Special} ->
       HasAt = lists:member($@, Special),
+      HasBangOrQuestionMark = lists:member($?, Special) orelse lists:member($!, Special),
```

Comment:
I'm not confident this is the best method of finding out if an identifier ends with `?` or `!`.

Reply:
Maybe lists:last would be better here?

Reply:
@michalmuskala I started with that but `Special` would be `"!@"` for `foo@bar!` IIRC (I tried earlier and I think this was the result).